HelgaShiryaeva/assignment1/Anagram.java

```diff
@@ -0,0 +1,74 @@
+package assignment1;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.StringTokenizer;
+
+public class Anagram {
+    private static final String PUNCTUATION_MARKS = " ,?!.:";
+
+    /**
+     * Checks if each word from the first string is an anagram of one of the words
+     * from the second one. Strings are considered to be anagrams this way.
+     * Supports punctuation marks : ?!,.:
+     * @param first is the first string to be checked if it is an anagram of
+     * @param second string
+     * @return true if each word from the first string is an anagram of one
+     * of the words from the second string
+     */
+    public boolean areAnagramsByWords(String first, String second) {
+        if (first.length() != second.length()) {
+            return false;
+        }
+        StringTokenizer stringTokenizerFirst = new StringTokenizer(first, PUNCTUATION_MARKS );
+        StringTokenizer stringTokenizerSecond = new StringTokenizer(second, PUNCTUATION_MARKS);
+        if (stringTokenizerFirst.countTokens() != stringTokenizerSecond.countTokens()) {
+            return false;
+        }
+        Map<Character, Integer> characterCountMap = new HashMap<>();
```

Comment:
`characterCountMap` is used only inside `areWordsAnagrams`. No need to create it outside that method and pass as parameter. Just create this map in the method where you use it.